### PR TITLE
Muzzle code generation cleanup

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**
- remove hardcoded class names from `MuzzleCodeGenerator` (easier renaming in the future)
- store field/method descriptors in references instead of types/lists of types
- remove unnecessary primitive type comparison in `ReferenceMatcher` (comparing descriptors is enough)
- improve `printMuzzleReferences` output

Example `printMuzzleReferences` output:

```
  ClassRef: io.opentelemetry.instrumentation.servlet.v3_0.Servlet3Accessor$Listener extends java.lang.Object implements javax.servlet.AsyncListener
    Sources:
      at: io.opentelemetry.instrumentation.servlet.v3_0.Servlet3Accessor:33
      at: io.opentelemetry.instrumentation.servlet.v3_0.Servlet3Accessor$Listener:0
      at: io.opentelemetry.instrumentation.servlet.v3_0.Servlet3Accessor$Listener:50
      at: io.opentelemetry.instrumentation.servlet.v3_0.Servlet3Accessor$Listener:55
      at: io.opentelemetry.instrumentation.servlet.v3_0.Servlet3Accessor$Listener:60
      at: io.opentelemetry.instrumentation.servlet.v3_0.Servlet3Accessor$Listener:65
      at: io.opentelemetry.instrumentation.servlet.v3_0.Servlet3Accessor$Listener:46
    FieldRef: PRIVATE_OR_HIGHER NON_STATIC io.opentelemetry.instrumentation.servlet.ServletAsyncListener listener
    MethodRef: PUBLIC NON_STATIC NON_FINAL void onComplete(javax.servlet.AsyncEvent)
    MethodRef: PUBLIC NON_STATIC NON_FINAL void onTimeout(javax.servlet.AsyncEvent)
    MethodRef: PUBLIC NON_STATIC NON_FINAL void onError(javax.servlet.AsyncEvent)
    MethodRef: PUBLIC NON_STATIC NON_FINAL void onStartAsync(javax.servlet.AsyncEvent)
```

### CODE REVIEW

File: javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/collector/MuzzleCodeGenerator.java [hunk 1]
```diff
@@ -378,12 +380,12 @@ private void generateMuzzleReferencesMethod(ReferenceCollector collector) {
             mv.visitLdcInsn(field.getName());
 
             { // field type
-              mv.visitLdcInsn(field.getType().getDescriptor());
+              mv.visitLdcInsn(field.getDescriptor());
               mv.visitMethodInsn(
                   Opcodes.INVOKESTATIC,
-                  Type.getInternalName(Type.class),
+                  typeType.getInternalName(),
```
Comment (trask) [inline javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/collector/MuzzleCodeGenerator.java:386]: 😁 👍

File: javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/collector/MuzzleCodeGenerator.java [hunk 2]
```diff
@@ -445,38 +444,43 @@ private void generateMuzzleReferencesMethod(ReferenceCollector collector) {
 
             mv.visitLdcInsn(method.getName());
 
-            { // return type
-              mv.visitLdcInsn(method.getReturnType().getDescriptor());
+            {
+              // method return and parameter types must be passed by Type.getType() call to be
+              // properly shaded
```
Comment (trask) [inline javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/collector/MuzzleCodeGenerator.java:449]: I don't understand this comment

File: javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/matcher/ReferenceMatcher.java [hunk 3]
```diff
@@ -291,42 +254,19 @@ private static void collectMethodsFromTypeHierarchy(
...
-          && ((fieldType
-                  .getType()
-                  .asErasure()
-                  .getInternalName()
-                  .equals(fieldRef.getType().getInternalName()))
-              || (fieldType.getType().asErasure().isPrimitive()
-                  && matchesPrimitive(
-                      fieldType.getType().asErasure().getInternalName(),
-                      fieldRef.getType().getInternalName())))) {
+          && fieldType.getDescriptor().equals(fieldRef.getDescriptor())) {
```
Comment (trask) [inline javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/matcher/ReferenceMatcher.java:269]: 💯

**Review by trask — APPROVED** [review]

File: javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/collector/MuzzleCodeGenerator.java (hunk 2 above)
Comment (mateuszrzeszutek) [reply javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/collector/MuzzleCodeGenerator.java:449]: Oh, I tried just passing the method descriptor string, but then it turned out that the shading plugin does not treat it as package/class name - so I had to stay with `Type`s. I'll reword that comment.

**Pull request merged**